src/k8s/pkg/utils/certificate.go

```diff
@@ -26,3 +32,75 @@ func SplitIPAndDNSSANs(extraSANs []string) ([]net.IP, []string) {
 
 	return ipSANs, dnsSANs
 }
+
+func TLSClientConfig(remoteCert *x509.Certificate) (*tls.Config, error) {
+	if remoteCert == nil {
+		return nil, fmt.Errorf("invalid remote public key")
+	}
+
+	config := &tls.Config{}
+
+	// Add the public key to the CA pool to make it trusted.
+	config.RootCAs = x509.NewCertPool()
+	remoteCert.IsCA = true
+	remoteCert.KeyUsage = x509.KeyUsageCertSign
+	config.RootCAs.AddCert(remoteCert)
+
```

Comment:
how much of this is required? let's not allow more than the absolute minimum necessary things. Also, `x509.NewCertPool()` ignores the root CAs from the host system, which _might_ be a problem if we use this elsewhere.